src/inferencetensor.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+//*****************************************************************************
+// Copyright 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <memory>
+#include <optional>
+#include <string>
+
+#include "pocapi.hpp"
+#include "shape.hpp"
+
+namespace ovms {
+class Buffer;
+class Status;
+
+class InferenceTensor {
+    const DataType datatype;
+    shape_t shape;
+    std::unique_ptr<Buffer> buffer;
+
+public:
+    InferenceTensor(DataType datatype, const size_t* shape, size_t dimCount);
+    ~InferenceTensor();
+    InferenceTensor(InferenceTensor&&);
+    InferenceTensor(const InferenceTensor&) = delete;
+    InferenceTensor& operator=(const InferenceTensor&) = delete;
+    InferenceTensor& operator=(const InferenceTensor&&);
+    Status setBuffer(const void* addr, size_t byteSize, BufferType bufferType, std::optional<uint32_t> deviceId);
+    Status removeBuffer();
+    DataType getDataType() const;
+    const shape_t& getShape() const;
```

Comment:
Here the shape is vector but to add shape you pass size_t* and dimCount, can we unify?

Reply:
This won't be exposed via C-API

Reply:
This should be used in internal OVMS code